pkg/kubelet/apis/podresources/server.go

```diff
@@ -79,7 +83,60 @@ func (p *podResourcesServer) List(ctx context.Context, req *v1alpha1.ListPodReso
 		podResources[i] = &pRes
 	}
 
-	return &v1alpha1.ListPodResourcesResponse{
+	return &podresourcesapi.ListPodResourcesResponse{
 		PodResources: podResources,
 	}, nil
 }
+
+type numaIDs []int64
+
+func (a numaIDs) Len() int           { return len(a) }
+func (a numaIDs) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a numaIDs) Less(i, j int) bool { return a[i] < a[j] }
+
+// GetAllocatableResources returns information about all the devices known by the server
+func (p *podResourcesServer) GetAllocatableResources(context.Context, *podresourcesapi.AllocatableResourcesRequest) (*podresourcesapi.AllocatableResourcesResponse, error) {
+	allDevices := p.devicesProvider.GetAllDevices()
+	var respDevs []*podresourcesapi.ContainerDevices
+
+	resourceNames := []string{}
+	for resourceName := range allDevices {
+		resourceNames = append(resourceNames, resourceName)
+	}
+	sort.Strings(resourceNames)
```

Comment:
We might need to specify the order (in the KEP), or if we don't guarantee it, specify in the API description that the order is not guaranteed.

Reply:
Good point. Both approaches have pros and cons, but I'm inclined towards NOT requiring the ordering in the API contract (thus in the KEP). I will update accordingly.

Reply:
Documented in https://github.com/kubernetes/enhancements/pull/2115